parser/Parser/Block.h

```diff
@@ -1,8 +1,16 @@
 #pragma once
-class Block
+#include <cstdint>
+#include "Reader.h"
+
+// not virtual! This might be counter-intuitive
+struct Block
 {
+public:
 	char ID;
-	unsigned int lengthOfBlock;
-
+	uint64_t lengthOfBlock;
+	void read(Reader& r) {
```

Comment:
TODO before merging, it would be wholesome to adding checks before we forget them.

E.g. is there enough data in the file? Is everything according to standard (e.g. `Something::header::tag` must not contain new-line character).